src/libraries/System.Private.CoreLib/src/System/IO/Enumeration/FileSystemEntry.Unix.cs

```diff
@@ -33,36 +33,6 @@ namespace System.IO.Enumeration
             entry._fullPath = ReadOnlySpan<char>.Empty;
             entry._fileName = ReadOnlySpan<char>.Empty;
             entry._status.InvalidateCaches();
-            entry._status.InitiallyDirectory = false;
```

Comment:
this property is called `InitiallyDirectory` which suggests that we want to know whether something was **initially** a directory. If we don't populate the property during init, the property looses the meaning?

@carlossanlop @Jozkee is that acceptable?

Reply:
I think we should move `InitiallyDirectory` out of `FileStatus` and into `FileSystemInfo` (which is where it is actually needed).
I can add such a change in this PR, or make it in a separate PR.

Reply:
`InitiallyDirectory` had an important job in preventing a huge perf regression when enumerating files.

Since we already had the `InodeType` information at this point, we could avoid an extra p/invoke inside this function to learn if the entry was a symlink to a directory or not.

The purpose of `InitiallyDirectory` was to learn from the beginning if the file was a symlink pointing to a directory. That way, if the target directory was deleted at some point, we would still know if the target was supposed to be a directory.

Can you please run the enumeration benchmarks, @tmds?

Reply:
Also, I don't recall if we already have a unit test for this, but can you manually verify the following?:

- From bash, create a directory with a subdirectory inside.
- Inside the parent directory, create a symlink that points to the subdirectory.
-  In a C# console app that consumes your changes, enumerate the contents of the parent directory, but make sure you return `FileSystemInfo` instances. Save the enumeration result.
- From bash, delete the subdirectory.
- Find inside your enumeration results the one representing the symlink.
- Without casting the `FileSystemInfo` that represents the symlink into a `DirectoryInfo`, invoke its method `ResolveLinkTarget`. It should [return a non-null](https://docs.microsoft.com/en-us/dotnet/api/system.io.filesysteminfo.resolvelinktarget?view=net-6.0#returns) `FileSystemInfo` instance that represents the now deleted subdirectory.
- Check `targetInfo.Exists`. The property was calling `FileStatus.GetExists()` which would compare the value of `_isDirectory` (it gets updated) with the original value of `InitiallyDirectory` which should've been set to `true` and never changed: 
 https://github.com/dotnet/runtime/blob/80ca504d085b4ff935fee8b011a2136cc75160fe/src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.Unix.cs#L261